patches/coreboot-4.8.1/0009-Add-heads-TPM-measurements-to-Skylake-Kabylake.patch

```diff
@@ -0,0 +1,181 @@
+diff --git ./src/arch/x86/acpi.c ./src/arch/x86/acpi.c
+index 8b6b2c1..fca4a76 100644
+--- ./src/arch/x86/acpi.c
++++ ./src/arch/x86/acpi.c
+@@ -48,6 +48,7 @@
+ #include <cpu/x86/lapic_def.h>
+ #include <cpu/cpu.h>
+ #include <cbfs.h>
++#include <security/tpm/tss.h>
+ 
+ u8 acpi_checksum(u8 *table, u32 length)
+ {
+@@ -1020,6 +1021,10 @@ unsigned long write_acpi_tables(unsigned long start)
+ 		return current;
+ 	}
+ 
++	if (IS_ENABLED(CONFIG_MEASURED_BOOT)) {
++	  tlcl_measure(2, (const void*) dsdt_file, dsdt_size);
++	}
++
+ 	slic_file = cbfs_boot_map_with_leak(CONFIG_CBFS_PREFIX "/slic",
+ 				     CBFS_TYPE_RAW, &slic_size);
+ 	if (slic_file
+diff --git ./src/arch/x86/postcar.c ./src/arch/x86/postcar.c
+index 6497b73..e846b69 100644
+--- ./src/arch/x86/postcar.c
++++ ./src/arch/x86/postcar.c
+@@ -19,6 +19,7 @@
+ #include <console/console.h>
+ #include <main_decl.h>
+ #include <program_loading.h>
++#include <security/tpm/tss.h>
+ #include <soc/intel/common/util.h>
+ 
+ /*
+@@ -43,3 +44,11 @@ void main(void)
+ 	/* Load and run ramstage. */
+ 	run_ramstage();
+ }
++
++void platform_segment_loaded(uintptr_t start, size_t size, int flags)
++{
++	if (IS_ENABLED(CONFIG_MEASURED_BOOT) && !(flags & SEG_NO_MEASURE)) {
++	  tlcl_measure(2, (const void*) start, size);
++	}
++}
++
+diff --git ./src/drivers/intel/fsp2_0/memory_init.c ./src/drivers/intel/fsp2_0/memory_init.c
+index 30987ce..124d3fa 100644
+--- ./src/drivers/intel/fsp2_0/memory_init.c
++++ ./src/drivers/intel/fsp2_0/memory_init.c
+@@ -150,10 +150,11 @@ static void do_fsp_post_memory_init(bool s3wake, uint32_t fsp_version)
+ 
+ 	/*
+ 	 * Initialize the TPM, unless the TPM was already initialized
+-	 * in verstage and used to verify romstage.
++	 * in verstage and used to verify romstage, or for measured boot.
+ 	 */
+ 	if (IS_ENABLED(CONFIG_LPC_TPM) &&
+-	    !IS_ENABLED(CONFIG_VBOOT_STARTS_IN_BOOTBLOCK))
++	    !IS_ENABLED(CONFIG_VBOOT_STARTS_IN_BOOTBLOCK) &&
++			!IS_ENABLED(CONFIG_MEASURED_BOOT))
+ 		init_tpm(s3wake);
+ }
+ 
+@@ -483,8 +484,33 @@ void fsp_memory_init(bool s3wake)
+ 	if (status != CB_SUCCESS)
+ 		die("Loading FSPM failed!\n");
+ 
++	if (IS_ENABLED(CONFIG_MEASURED_BOOT) && IS_ENABLED(CONFIG_LPC_TPM)) {
++		// we don't know if we are coming out of a resume
++		// at this point, but want to setup the tpm ASAP
++		init_tpm(0);
++		tlcl_lib_init();
++		const void * const bootblock = (const void*) 0xFFFFF800;
++		const unsigned bootblock_size = 0x800;
++		tlcl_measure(0, bootblock, bootblock_size);
++
++		tlcl_measure(1, _romstage, _eromstage - _romstage);
++	}
++
++	if (IS_ENABLED(CONFIG_MEASURED_BOOT)) {
++	  tlcl_measure(1, (const void*) hdr.image_base, hdr.image_size);
++	}
```

Comment:
Why are you doing this here ? First, it might as well go in the if() just above it if there is a justifitication of doing it this way.. but also, I don't see why it's measured separately.. because you're measuring the hdr.image_base into PCR 1 right here, and then on the next line, you add the SEG_NO_MEASURE so you don't measure hdr.image_base (but the platform_segment_loaded function defined just below is measuring into PCR 1, so it's not because you want it in PCR 1 instead of whatever the prog_segment_loaded measured it into). So what is the point of doing all that ?
I suggest you remove this and remove the SEG_NO_MEASURE from the next line.

Reply:
Sorry - this was leftover from when I thought segments_loading here were for ramstage (I forgot about postcar on this platform) so I wanted it to specifically be in PCR1 (instead of PCR2). Reverting